pennylane/beta/tapes/cv_param_shift.py

```diff
@@ -0,0 +1,458 @@
+            params (list[Any]): the quantum tape operation parameters
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        # pylint: disable=protected-access
+
+        if "PolyXP" not in device.observables:
+            # If the device does not support PolyXP, must fallback
+            # to numeric differentiation.
+            warnings.warn(
+                f"The device {device.short_name} does not support "
+                "the PolyXP observable. The analytic parameter-shift cannot be used for "
+                "second-order observables; falling back to finite-differences.",
+                UserWarning,
+            )
+            return self.numeric_pd(idx, device, params, **options)
+
+        t_idx = list(self.trainable_params)[idx]
+        op = self._par_info[t_idx]["op"]
+        p_idx = self._par_info[t_idx]["p_idx"]
+
+        recipe = op.grad_recipe[p_idx]
+        c, s = (0.5, np.pi / 2) if recipe is None else recipe
+
+        shift = np.zeros_like(params)
+        shift[idx] = s
+
+        # evaluate transformed observables at the original parameter point
+        # first build the Heisenberg picture transformation matrix Z
+        self.set_parameters(params + shift)
+        Z2 = op.heisenberg_tr(device.wires)
+
+        self.set_parameters(params - shift)
+        Z1 = op.heisenberg_tr(device.wires)
+
+        # derivative of the operation
+        Z = (Z2 - Z1) * c
+
+        self.set_parameters(params)
+        Z0 = op.heisenberg_tr(device.wires, inverse=True)
+        Z = Z @ Z0
+
+        # conjugate Z with all the descendant operations
+        B = np.eye(1 + 2 * device.num_wires)
+        B_inv = B.copy()
+
+        succ = self.graph.descendants_in_order((op,))
+        operation_descendents = itertools.filterfalse(qml.circuit_graph._is_observable, succ)
+        observable_descendents = filter(qml.circuit_graph._is_observable, succ)
+
+        for BB in operation_descendents:
+            if not BB.supports_heisenberg:
+                # if the descendant gate is non-Gaussian in parameter-shift differentiation
+                # mode, then there must be no observable following it.
+                continue
+
+            B = BB.heisenberg_tr(device.wires) @ B
+            B_inv = B_inv @ BB.heisenberg_tr(device.wires, inverse=True)
+
+        Z = B @ Z @ B_inv  # conjugation
+
+        # transform the descendant observables into their derivatives using Z
+        transformed_obs_idx = []
+
+        for obs in observable_descendents:
+            # get the index of the descendent observable
+            idx = self.observables.index(obs)
+            transformed_obs_idx.append(idx)
+
+            # transform the observable
+            self._measurements[idx] = MeasurementProcess(
+                qml.operation.Expectation, self._transform_observable(obs, Z, device.wires)

```

Comment:
Note: there _could_ be a case where the transformed observable is of the form
```python
PolyXP(np.array([c, 0, 0, 0, 0...]), wires=...)
```

That is, only the identity term is non-zero. On an ideal device (with no cutoff), we could replace the device evaluation here simply with `c` (since `<PolyXP(np.array([c, 0, 0, 0, 0...]), wires=...)> = c`).

This puts us in an interesting position, where we are calculating the analytic gradient *efficiently*, *without needing to query the hardware*.

Reply:
Have updated the code to include this as a todo